src/main/java/seedu/address/model/task/DistinctTaskList.java

```diff
@@ -20,14 +21,26 @@ public class DistinctTaskList implements Iterable<Task> {
     public final ObservableList<Task> unmodifiableTaskList = FXCollections
             .unmodifiableObservableList(taskList);
 
+    /**
+     * Returns true if the list contains an equivalent task as the given argument.
+     */
+    public boolean contains(Task toCheck) {
+        requireNonNull(toCheck);
+        return taskList.stream().anyMatch(toCheck::isSameTask);
+    }
+
     /**
      * Adds the task to the taskList.
+     * The task must not already exist in the list.
      *
      * @param taskAdded The task to be added.
      */
     public void addTask(Task taskAdded) {
         requireNonNull(taskAdded);
         //Might have to add address book as param to check if addressBook already has module
+        if (contains(taskAdded)) {
```

Comment:
Maybe you could add the part to check whether the existing module list in addressbook contains the module code so that tasks with invalid module code cannot be added